http-server-netty/src/main/java/io/micronaut/http/server/netty/DefaultFilterBodyParser.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.netty;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.async.annotation.SingleResult;
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.form.FormUrlEncodedDecoder;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.ServerHttpRequest;
+import io.micronaut.http.body.ByteBody;
+import io.micronaut.http.body.CloseableByteBody;
+import io.micronaut.http.filter.FilterBodyParser;
+import io.micronaut.json.JsonMapper;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+/**
+ * Implementation of {@link FilterBodyParser} which leverages {@link ServerHttpRequest#byteBody()} API.
+ * For form-url-encoded payloads it uses the {@link FormUrlEncodedDecoder} API.
+ * For JSON payloads it uses a {@link JsonMapper}.
+ * @author Sergio del Amo
+ * @since 4.11.0
+ */
+@Requires(beans = {JsonMapper.class, FormUrlEncodedDecoder.class})
+@Singleton
+@Experimental
+final class DefaultFilterBodyParser implements FilterBodyParser {
```

Comment:
why is this still part of `http-server-netty`?

Reply:
A mistake. [I moved them to `http-server`](https://github.com/micronaut-projects/micronaut-core/pull/11272/commits/0adaccfcdbdeb74af26e933cd438c38979a2b23b).